developer_tools/minprintf.h

```diff
@@ -0,0 +1,352 @@
+/**
+ * @file
+ * @brief Implementation of a function similar to printf
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂